src/neuroconv/datainterfaces/behavior/deeplabcut/deeplabcutdatainterface.py

```diff
@@ -30,21 +32,89 @@ def __init__(
         file_path: FilePath,
         config_file_path: FilePath | None = None,
         subject_name: str = "ind1",
+        pose_estimation_metadata_key: str = "PoseEstimationDeepLabCut",
         verbose: bool = False,
     ):
         """
-        Interface for writing DLC's output files to nwb using dlc2nwb.
+        Interface for writing DeepLabCut's output files to NWB.
+
+        This interface reads DeepLabCut output files (.h5 or .csv) and converts them to NWB format
+        using the ndx-pose extension. It extracts keypoints (bodyparts), their coordinates, and confidence
+        values, and organizes them into a structured format within the NWB file.
 
         Parameters
         ----------
         file_path : FilePath
-            Path to the file output by dlc (.h5 or .csv).
+            Path to the file output by DeepLabCut (.h5 or .csv). The file should contain the pose estimation
+            data with keypoints, coordinates, and confidence values.
         config_file_path : FilePath, optional
-            Path to .yml config file
+            Path to the DeepLabCut .yml config file. If provided, additional metadata such as video dimensions,
+            task description, and experimenter information will be extracted.
         subject_name : str, default: "ind1"
-            The name of the subject for which the :py:class:`~pynwb.file.NWBFile` is to be created.
-        verbose: bool, default: True
-            Controls verbosity.
+            The subject name to be used in the metadata. For output files with multiple individuals,
+            this must match the name of the individual for which the data will be added. This name is also
+            used to link the skeleton to the subject in the NWB file.
+        pose_estimation_metadata_key : str, default: "PoseEstimationDeepLabCut"
+            This controls where in the metadata the pose estimation metadata is stored:
+            metadata["PoseEstimation"]["PoseEstimationContainers"][pose_estimation_metadata_key].
+            This key is also used as the name of the PoseEstimation container in the NWB file.
+        verbose : bool, default: False
+            Controls verbosity of the conversion process.
+
+
+        Metadata Structure
+        ------------------
+        The metadata is organized in a hierarchical structure:
+
+        metadata["PoseEstimation"]["Skeletons"][skeleton_name] contains:
+        {
+            "name": "SkeletonPoseEstimationDeepLabCut_SubjectName",
+            "nodes": ["bodypart1", "bodypart2", ...],  # List of keypoints/bodyparts
+            "edges": [[0, 1], [1, 2], ...],  # Connections between nodes (optional)
```

Comment:
These seem more like data than metadata to me.

Reply:
I think that's a good point especially with the edges.

With the nodes, I think it is nice being able to edit them. With the edges, I feel it is OK to leave it as it just as an escape valve for when users want a fix with this.

How do you think about it? What do you think is better?

Reply:
If you renamed the nodes in the metadata, would the interface even work? Based on my reading of `_write_pes_to_nwbfile`, it seems like the node names must match the values in the data.

Reply:
Let me re-read this. Which line is where the clash will occur?

Reply:
Ah, no, this should be fine actually, my bad. I was conflating keypoints with nodes (by default they are the same), but actually they are distinct.

Reply:
Yeah after more consideration, I think this is fine to expose.